### ISSUE TITLE
Add location sharing feature (Element parity) [Low priority]

### ISSUE THREAD

**Issue by Extarys** [ajbura/cinny#446]
### Describe the problem

It might be hard to safely and rapidly communicate someone's location to a friend or a group of people while on the move.

### Describe the solution you'd like

Ability to share either a static or a real time location information to a room.

### Additional context

See element blog post about this feature: https://element.io/blog/element-launches-e2ee-location-sharing/

Some issues/to-dos I found on the element-web GH for inspiration:

- [ ] Create a way to access more "send" options (see image 1). Three-dot menu next to the current emoji would be "standard"
- [ ] Must be sent/stored E2EE
- [ ] For real time location: Set expiration time/date, at which point the location will no longer be updated
- [ ] Issue to avoid: Don't crash if location cannot be found (Permission not granted, GPS/position not yet known)
- [ ] Issue to avoid: Don't forget to resize the map depending on the device/available width

[See all location sharing issues from element-web](https://github.com/vector-im/element-web/issues?q=+label%3AA-Location-Sharing+)

**image 1:**
![image](https://user-images.githubusercontent.com/11581433/160727963-69d4417a-a8a7-457d-afed-c52cc21f8e02.png)

I started using cinny today, and I prefer how the interface is managed compared to element (and the themes are lovely!), but, of course, features are missing. I wouldn't mind trying to document some of the candies that are missing and create issues related to them and expand if that isn't a burden to the dev/contributors. My goal wouldn't to overwhelm you, but to have a clear view of what has been done and what is missing so new and unfamiliar users know what is going on.

**Comment by Neved4** [comment]
Would love to sort this on [cinny-desktop](https://github.com/cinnyapp/cinny-desktop) when it's sorted out on [cinny](https://github.com/cinnyapp/cinny) web.

For reference, it'd would be done by [tauri-plugin-geolocation](https://crates.io/crates/tauri-plugin-geolocation), but it seems to be a v2 tauri module.

Thus it's currently blocked by https://github.com/cinnyapp/cinny-desktop/issues/398

@kfiven tag this one too if you can! Cheers!
